### ISSUE TITLE
Drill-through not working on implicit joined tables with sandboxes

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#13641]
**Describe the bug**
Not possible for sandboxed user to drill-through when question is using implicit joins and grouped by a joined column.
Doesn't matter if field remapping to the joined table.

Workaround is to use explicit joins (manually joining tables in the question).

**To Reproduce**
1. Admin > People > create user "U1" with attribute `user_id=1`
2. Admin > Permissions > Sample Dataset > revoke all, sandbox Orders on "User ID"=`user_id`, and full access to Products
![image](https://user-images.githubusercontent.com/1447303/92602201-1c39c200-f2ae-11ea-82ae-b14401a8682e.png)
3. Custom question > Sample Dataset > Orders - summarize Count by Category, save as "Q1"
![image](https://user-images.githubusercontent.com/1447303/97719973-ff5c8680-1ac7-11eb-8c9e-c9318ceaaaaf.png)
4. Login as "U1" > open "Q1" > click one of the bars to "View these Orders" - fails with `Column "source.CATEGORY" not found;`

<details><summary>Full stacktrace</summary>

```
2020-10-30 16:54:22,850 ERROR middleware.catch-exceptions :: Error processing query: null
{:database_id 1,
 :started_at #t "2020-10-30T16:54:22.168199+01:00[Europe/Copenhagen]",
 :state "42S22",
 :json_query
 {:type "query", :query {:source-table 2, :filter ["and" ["=" ["field-id" 6] "Gadget"]]}, :database 1, :parameters [], :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class org.h2.jdbc.JdbcSQLException,
 :stacktrace
 ["org.h2.message.DbException.getJdbcSQLException(DbException.java:357)"
  "org.h2.message.DbException.get(DbException.java:179)"
  "org.h2.message.DbException.get(DbException.java:155)"
  "org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)"
  "org.h2.expression.Comparison.optimize(Comparison.java:177)"
  "org.h2.command.dml.Select.prepare(Select.java:861)"
  "org.h2.command.Parser.prepareCommand(Parser.java:283)"
  "org.h2.engine.Session.prepareLocal(Session.java:611)"
  "org.h2.engine.Session.prepareCommand(Session.java:549)"
  "org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)"
  "org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)"
  "org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:1135)"
  "com.mchange.v2.c3p0.impl.NewProxyConnection.prepareStatement(NewProxyConnection.java:267)"
  "--> driver.sql_jdbc.execute$fn__44809.invokeStatic(execute.clj:239)"
  "driver.sql_jdbc.execute$fn__44809.invoke(execute.clj:237)"
  "driver.sql_jdbc.execute$prepared_statement_STAR_.invokeStatic(execute.clj:257)"
  "driver.sql_jdbc.execute$prepared_statement_STAR_.invoke(execute.clj:254)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:390)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:377)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:386)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:377)"
  "driver.sql_jdbc$fn__82086.invokeStatic(sql_jdbc.clj:49)"
  "driver.sql_jdbc$fn__82086.invoke(sql_jdbc.clj:47)"
  "driver.h2$fn__80898.invokeStatic(h2.clj:87)"
  "driver.h2$fn__80898.invoke(h2.clj:84)"
  "query_processor.context$executef.invokeStatic(context.clj:59)"
  "query_processor.context$executef.invoke(context.clj:48)"
  "query_processor.context.default$default_runf.invokeStatic(default.clj:69)"
  "query_processor.context.default$default_runf.invoke(default.clj:67)"
  "query_processor.context$runf.invokeStatic(context.clj:45)"
  "query_processor.context$runf.invoke(context.clj:39)"
  "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
  "query_processor.reducible$pivot.invoke(reducible.clj:31)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__49316.invoke(mbql_to_native.clj:26)"
  "query_processor.middleware.check_features$check_features$fn__48591.invoke(check_features.clj:42)"
  "query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__49481.invoke(optimize_datetime_filters.clj:133)"
  "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__47392.invoke(auto_parse_filter_values.clj:44)"
  "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__43451.invoke(wrap_value_literals.clj:149)"
  "query_processor.middleware.annotate$add_column_info$fn__43314.invoke(annotate.clj:575)"
  "query_processor.middleware.permissions$check_query_permissions$fn__48466.invoke(permissions.clj:70)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__49999.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__48664.invoke(cumulative_aggregations.clj:61)"
  "metabase_enterprise.sandbox.query_processor.middleware.row_level_restrictions$apply_row_level_permissions$fn__51266.invoke(row_level_restrictions.clj:292)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__50236.invoke(resolve_joined_fields.clj:36)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__50555.invoke(resolve_joins.clj:183)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__46997.invoke(add_implicit_joins.clj:254)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__49277.invoke(large_int_id.clj:44)"
  "query_processor.middleware.limit$limit$fn__49302.invoke(limit.clj:38)"
  "query_processor.middleware.format_rows$format_rows$fn__49257.invoke(format_rows.clj:84)"
  "query_processor.middleware.desugar$desugar$fn__48730.invoke(desugar.clj:22)"
  "query_processor.middleware.binning$update_binning_strategy$fn__47751.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__48268.invoke(resolve_fields.clj:24)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__46545.invoke(add_dimension_projections.clj:318)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__46748.invoke(add_implicit_clauses.clj:141)"
  "metabase_enterprise.sandbox.query_processor.middleware.row_level_restrictions$apply_row_level_permissions$fn__51266.invoke(row_level_restrictions.clj:287)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__47146.invoke(add_source_metadata.clj:105)"
  "metabase_enterprise.sandbox.query_processor.middleware.column_level_perms_check$maybe_apply_column_level_perms_check$fn__50968.invoke(column_level_perms_check.clj:24)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__50196.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__47339.invoke(auto_bucket_datetimes.clj:125)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__48315.invoke(resolve_source_table.clj:46)"
  "query_processor.middleware.parameters$substitute_parameters$fn__49981.invoke(parameters.clj:114)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__48367.invoke(resolve_referenced.clj:80)"
  "query_processor.middleware.expand_macros$expand_macros$fn__48986.invoke(expand_macros.clj:158)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__47155.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__50917.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__50207$fn__50211.invoke(resolve_database_and_driver.clj:33)"
  "driver$do_with_driver.invokeStatic(driver.clj:61)"
  "driver$do_with_driver.invoke(driver.clj:57)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__50207.invoke(resolve_database_and_driver.clj:27)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49204.invoke(fetch_source_query.clj:267)"
  "query_processor.middleware.store$initialize_store$fn__50926$fn__50927.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__50926.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__48244.invoke(cache.clj:214)"
  "query_processor.middleware.validate$validate_query$fn__50935.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__49329.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__47015.invoke(add_rows_truncated.clj:36)"
  "metabase_enterprise.audit.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__31552.invoke(handle_audit_queries.clj:165)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__50902.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__48607.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__50070.invoke(process_userland_query.clj:136)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__48550.invoke(catch_exceptions.clj:174)"
  "query_processor.reducible$async_qp$qp_STAR___40109$thunk__40110.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___40109.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___40118$fn__40121.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___40118.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:237)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:233)"
  "query_processor$fn__51311$process_query_and_save_execution_BANG___51320$fn__51323.invoke(query_processor.clj:249)"
  "query_processor$fn__51311$process_query_and_save_execution_BANG___51320.invoke(query_processor.clj:241)"
  "query_processor$fn__51355$process_query_and_save_with_max_results_constraints_BANG___51364$fn__51367.invoke(query_processor.clj:261)"
  "query_processor$fn__51355$process_query_and_save_with_max_results_constraints_BANG___51364.invoke(query_processor.clj:254)"
  "api.dataset$fn__62550$fn__62553.invoke(dataset.clj:59)"
  "query_processor.streaming$streaming_response_STAR_$fn__37118$fn__37119.invoke(streaming.clj:73)"
  "query_processor.streaming$streaming_response_STAR_$fn__37118.invoke(streaming.clj:72)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)"
  "async.streaming_response$do_f_async$fn__24642.invoke(streaming_response.clj:85)"],
 :context :ad-hoc,
 :error
 "Column \"source.CATEGORY\" not found; SQL statement:\n-- Metabase:: userID: 2 queryType: MBQL queryHash: 8ea2371b4dfbb51f69cf6f042debf36a013574b2a9dac257bc859fed22b1d03d\nSELECT \"source\".\"ID\" AS \"ID\", \"source\".\"USER_ID\" AS \"USER_ID\", \"source\".\"PRODUCT_ID\" AS \"PRODUCT_ID\", \"source\".\"SUBTOTAL\" AS \"SUBTOTAL\", \"source\".\"TAX\" AS \"TAX\", \"source\".\"TOTAL\" AS \"TOTAL\", \"source\".\"DISCOUNT\" AS \"DISCOUNT\", \"source\".\"CREATED_AT\" AS \"CREATED_AT\", \"source\".\"QUANTITY\" AS \"QUANTITY\" FROM (SELECT \"PUBLIC\".\"ORDERS\".\"ID\" AS \"ID\", \"PUBLIC\".\"ORDERS\".\"USER_ID\" AS \"USER_ID\", \"PUBLIC\".\"ORDERS\".\"PRODUCT_ID\" AS \"PRODUCT_ID\", \"PUBLIC\".\"ORDERS\".\"SUBTOTAL\" AS \"SUBTOTAL\", \"PUBLIC\".\"ORDERS\".\"TAX\" AS \"TAX\", \"PUBLIC\".\"ORDERS\".\"TOTAL\" AS \"TOTAL\", \"PUBLIC\".\"ORDERS\".\"DISCOUNT\" AS \"DISCOUNT\", \"PUBLIC\".\"ORDERS\".\"CREATED_AT\" AS \"CREATED_AT\", \"PUBLIC\".\"ORDERS\".\"QUANTITY\" AS \"QUANTITY\" FROM \"PUBLIC\".\"ORDERS\" WHERE \"PUBLIC\".\"ORDERS\".\"USER_ID\" = 1) \"source\" WHERE \"source\".\"CATEGORY\" = ? LIMIT 2000 [42122-197]",
 :row_count 0,
 :running_time 0,
 :preprocessed
 {:type :query,
  :query
  {:filter [:= [:field-id 6] [:value "Gadget" {:base_type :type/Text, :special_type :type/Category, :database_type "VARCHAR", :name "CATEGORY"}]],
   :source-metadata
   [{:name "ID", :id 12, :table_id 2, :display_name "ID", :base_type :type/BigInteger, :special_type :type/PK, :fingerprint nil, :settings nil}
    {:name "USER_ID", :id 9, :table_id 2, :display_name "User ID", :base_type :type/Integer, :special_type :type/FK, :fingerprint {:global {:distinct-count 929, :nil% 0.0}}, :settings {}}
    {:name "PRODUCT_ID", :id 11, :table_id 2, :display_name "Product ID", :base_type :type/Integer, :special_type :type/FK, :fingerprint {:global {:distinct-count 200, :nil% 0.0}}, :settings {}}
    {:name "SUBTOTAL",
     :id 13,
     :table_id 2,
     :display_name "Subtotal",
     :base_type :type/Float,
     :special_type nil,
     :fingerprint
     {:global {:distinct-count 340, :nil% 0.0},
      :type {:type/Number {:min 15.691943673970439, :max 148.22900526552291, :avg 77.01295465356547, :sd 32.53705013056317, :q1 49.74894519060184, :q3 105.42965746993103}}},
     :settings nil}
    {:name "TAX",
     :id 16,
     :table_id 2,
     :display_name "Tax",
     :base_type :type/Float,
     :special_type nil,
     :fingerprint
     {:global {:distinct-count 797, :nil% 0.0}, :type {:type/Number {:min 0.0, :max 11.12, :avg 3.8722100000000004, :sd 2.3206651358900316, :q1 2.273340386603857, :q3 5.337275338216307}}},
     :settings nil}
    {:name "TOTAL",
     :id 17,
     :table_id 2,
     :display_name "Total",
     :base_type :type/Float,
     :special_type nil,
     :fingerprint
     {:global {:distinct-count 10000, :nil% 0.0},
      :type {:type/Number {:min 12.061602936923117, :max 238.32732001721533, :avg 82.96014815230805, :sd 38.35967664847571, :q1 52.006147617878135, :q3 109.55803018499738}}},
     :settings nil}
    {:name "DISCOUNT",
     :id 10,
     :table_id 2,
     :display_name "Discount",
     :base_type :type/Float,
     :special_type :type/Discount,
     :fingerprint
     {:global {:distinct-count 701, :nil% 0.898},
      :type {:type/Number {:min 0.17088996672584322, :max 61.69684269960571, :avg 5.161255547580326, :sd 3.053663125001991, :q1 2.9786226681458743, :q3 7.338187788658235}}},
     :settings nil}
    {:table_id 2,
     :special_type :type/CreationTimestamp,
     :unit :default,
     :name "CREATED_AT",
     :settings nil,
     :id 15,
     :display_name "Created At",
     :fingerprint {:global {:distinct-count 9998, :nil% 0.0}, :type {:type/DateTime {:earliest "2016-04-30T18:56:13.352", :latest "2020-04-19T14:07:15.657"}}},
     :base_type :type/DateTime}
    {:name "QUANTITY",
     :id 14,
     :table_id 2,
     :display_name "Quantity",
     :base_type :type/Integer,
     :special_type :type/Quantity,
     :fingerprint {:global {:distinct-count 62, :nil% 0.0}, :type {:type/Number {:min 0.0, :max 100.0, :avg 3.7015, :sd 4.214258386403798, :q1 1.755882607764982, :q3 4.882654507928044}}},
     :settings nil}],
   :fields
   [[:field-literal "ID" :type/BigInteger]
    [:field-literal "USER_ID" :type/Integer]
    [:field-literal "PRODUCT_ID" :type/Integer]
    [:field-literal "SUBTOTAL" :type/Float]
    [:field-literal "TAX" :type/Float]
    [:field-literal "TOTAL" :type/Float]
    [:field-literal "DISCOUNT" :type/Float]
    [:field-literal "CREATED_AT" :type/DateTime]
    [:field-literal "QUANTITY" :type/Integer]],
   :limit 2000,
   :source-query
   {:source-table 2,
    :filter [:= [:field-id 9] [:value 1 {:base_type :type/Integer, :special_type :type/FK, :database_type "INTEGER", :name "USER_ID"}]],
    :fields [[:field-id 12] [:field-id 9] [:field-id 11] [:field-id 13] [:field-id 16] [:field-id 17] [:field-id 10] [:datetime-field [:field-id 15] :default] [:field-id 14]],
    :gtap? true}},
  :database 1,
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
  :info
  {:executed-by 2,
   :context :ad-hoc,
   :nested? false,
   :query-hash [-114, -94, 55, 27, 77, -5, -75, 31, 105, -49, 111, 4, 45, -21, -13, 106, 1, 53, 116, -78, -87, -38, -62, 87, -68, -123, -97, -19, 34, -79, -48, 61]},
  :constraints {:max-results 10000, :max-results-bare-rows 2000},
  :gtap-perms #{"/db/1/schema/PUBLIC/table/2/query/"}},
 :data {:rows [], :cols []}}

2020-10-30 16:54:22,854 DEBUG middleware.log :: POST /api/dataset 202 [ASYNC: completed] 695.6 ms (22 DB calls) App DB connections: 0/7 Jetty threads: 3/50 (1 idle, 0 queued) (52 total active threads) Queries in flight: 0 (0 queued)
```

</details>

**Information about your Metabase Installation:**
Metabase 1.37.0.2

**Additional context**
Related to https://github.com/metabase/metabase-enterprise/issues/535

**Comment by camsaul** [comment]
The `Column "source.CATEGORY" not found; ` bug should be fixed by #14695 I think (I'll confirm)

**Comment by camsaul** [comment]
Fixed by #14719
